ports/unix/variants/COVERAGE/mpconfigvariant.mk

```diff
@@ -0,0 +1,24 @@
+PROG ?= micropython_coverage
+
+COPT = -O0
+
+CFLAGS_EXTRA += \
+	-fprofile-arcs -ftest-coverage \
+	-Wdouble-promotion -Wformat -Wmissing-declarations -Wmissing-prototypes -Wsign-compare \
+	-Wold-style-definition -Wpointer-arith -Wshadow -Wuninitialized -Wunused-parameter \
+	-DMICROPY_UNIX_COVERAGE
+
+LDFLAGS_EXTRA += -fprofile-arcs -ftest-coverage
+
+FROZEN_MANIFEST = manifest_coverage.py
+
+test: $(PROG) $(TOP)/tests/run-tests
```

Comment:
I think the test targets should remain in the main Makefile so they can be run on any variant.  Eg there'd be `test` and `test_full` (latter has all the lines below).  Then on travis coverage it'd be something like `make VARIANT=coverage test_full`

Reply:
OK, yeah I wasn't sure about this -- I figured the test target only makes sense for a subset of variants (i.e. not minimal), and that on the coverage variant that the `test` target should be the coverage run.

I think perhaps adding a BASE variant might address this (will expand on this in another comment).

Reply:
The tests can be orthogonal to the variant: the current coverage test target just runs more tests, but they should run just fine (maybe some skipped tho) on any variant.  Similarly it might be useful to run a basic set of tests on the coverage variant.  At the moment it's not possible to easily run tests on anything but the standard and coverage builds, so defining various test targets in the main makefile would make this possible.

Note that the `gcov ...` lines in the current `coverage_test` would need to split out into a separate target, like `gcov`.

Reply:
Moved the `test` target back to the main Makefile. As suggested, we should instead focus on making the tests run without failures on any variant by detecting what features are enabled. (See #5227)